### PR TITLE
feat(Politicians): Added cabinet view structure to politicians

### PR DESCRIPTION
**Pull request by TheCornelius**
feature or bug? : Feature
changes to user:
- At /politicians page, users get to see an additional section of politicians, categorized by ministries involved, contains links.

under the hood changes:
- Modified Politicians component to include cabinet view, with use of a feature flag
- Added mock_cabinet data in mock_cabinet.js
- Added links to individual politician's page

### CODE REVIEW

**Comment by TheCornelius** [comment]
closes #257

File: src/api/mock_cabinet.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Cabinet.vue
```
Comment (nazreen) [inline src/api/mock_cabinet.js]: this file should be put inside `src/mocks/mock_cabinet.js`

File: src/api/mock_cabinet.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        secondary_politician_ids: [
+          '-LAEDr85AyrHvQOBeLJu',  // Abang Jo
+          '31ddGKthhfXp8onAIeQ2'  // Ahmad Faizal Azumu
+        ],
+        secondary_title: 'Deputy Minister'
+      }
+  ]
+}
+
+export { mock_cabinet };
```
Comment (nazreen) [inline src/api/mock_cabinet.js]: export default mock_cabinet

Comment (nazreen) [reply src/api/mock_cabinet.js]: // e.g. of named export
<script>
export { mock_cabinet } from './mock_cabinet'
</script>

// e.g. of default export
<script>
export mock_cabinet from './mock_cabinet'
</script>

default export is more appropriate

File: src/api/index.js [hunk 3]
```diff
@@ -35,6 +35,17 @@ async function getSingle (path, id) {
   }
 }
 
+// Checking development variable
+// console.log(process.env.NODE_ENV);
+if (process.env.NODE_ENV == "development") {
+  // TODO: Cabinet view function here 
+  async function getCabinet() {
+
```
Comment (nazreen) [inline src/api/index.js]: return mock_cabinet
...
return Promise.resolve(mock_cabinet)

File: src/views/Politicians.vue [hunk 4]
```diff
@@ -53,6 +91,15 @@ export default {
...
           .includes(this.search.toLowerCase());
       });
+    },
+    // CABINET VIEW CODE
+    showCabinet() {
+      return process.env.NODE_ENV == "development";
+      // return this.mock_cabinet;
+    },
+    // CABINET VIEW CODE
+    getCabinet() {
```
Comment (nazreen) [inline src/views/Politicians.vue]: async getCabinet() {..}
..

async getCabinet() {
  this.cabinet = await getCabinet()
}

Comment (nazreen) [reply src/views/Politicians.vue]: this should be under methods

**Review by nazreen — CHANGES_REQUESTED** [review]
1. create data property and init cabinet with empty object
2. create methods property and place getCabinet there
3. inside created, call this.getCabinet to set value of this.cabinet

**Review by nazreen — CHANGES_REQUESTED** [review]
inside Politicians.vue:
- getCabinet from `src/api/index` should be imported
- methods.getCabinet should be calling that api `geCabinet`
- `mock_cabinet` should not be imported here

in summary, the mocking should be done in `src/api/index` and not in `Politicians.vue`

**Comment by TheCornelius** [comment]
closes #257 #259

**Comment by TheCornelius** [comment]
closes #260

**Comment by nazreen** [comment]
looks good, thanks @TheCornelius !

**Comment by nazreen** [comment]
@TheCornelius I made several quick changes pre-merging:

- moved CabinetView to top of page rather than after 'All Politicians'
- removed the borders (fits our current styling better), hope that's alright
- added 'All Politicians' h1 to make clear that the section below Cabinet View is all politicians

**Comment by TheCornelius** [comment]
The new /politicians page looks pretty clean

**Comment by nazreen** [comment]
@TheCornelius glad to hear that :)

**Pull request merged**